### ISSUE TITLE
Issue running 1D example 

### ISSUE THREAD

**Issue by huckl3b3rry87** [JuliaMath/Interpolations.jl#123]
When I run:

```
using Interpolations
nx = 10
f(x) = sin(2pi/(nx-1) * (x-1))
xcoarse = `1:nx`
ycoarse = f(xcoarse);
xfine = 1:.1:xcoarse[end]
yfine = f(xfine);
yitp = interpolate(ycoarse, BSpline(Linear), OnGrid)
```

I get:

```
WARNING: sin{T <: Number}(x::AbstractArray{T}) is deprecated, use sin.(x) instead.
 in depwarn(::String, ::Symbol) at ./deprecated.jl:64
 in sin(::FloatRange{Float64}) at ./deprecated.jl:50
 in f(::UnitRange{Int64}) at ./In[1]:4
 in include_string(::String, ::String) at ./loading.jl:471
 in execute_request(::ZMQ.Socket, ::IJulia.Msg) at /home/febbo/.julia/v0.6/IJulia/src/execute_request.jl:177
 in eventloop(::ZMQ.Socket) at /home/febbo/.julia/v0.6/IJulia/src/eventloop.jl:8
 in (::IJulia.##9#15)() at ./task.jl:363
while loading In[1], in expression starting on line 7

LoadError: MethodError: Cannot `convert` an object of type Type{Interpolations.Linear} to an object of type Interpolations.BSpline{D<:Interpolations.Degree}
This may have arisen from a call to the constructor Interpolations.BSpline{D<:Interpolations.Degree}(...),
since type constructors fall back to convert methods.
while loading In[1], in expression starting on line 12

 in Interpolations.BSpline{D<:Interpolations.Degree}(::Type{T}) at ./sysimg.jl:53
```

I am on Ubuntu 16.04 and 

```
            _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.6.0-dev.508 (2016-09-06 20:36 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit b1f1525 (27 days old master)
|__/                   |  x86_64-linux-gnu
```

Should I checkout the master Interpolations PKG?

**Comment by huckl3b3rry87** [comment]
Does Interpolations.jl work with the new julia?

**Comment by timholy** [comment]
I haven't tried it, but to me "the new julia" means version 0.5. Version 0.6 is bleeding-edge (recommended only for users who contribute fixes to the packages they need).

**Comment by huckl3b3rry87** [comment]
@timholy fair enough, I went back to 0.5, but I am still having issues when I type:

```
using Interpolations
nx = 10
f(x) = sin(2pi/(nx-1) * (x-1))
xcoarse = 1:nx
ycoarse = f(xcoarse);
xfine = 1:.1:xcoarse[end]
yfine = f(xfine);
yitp = interpolate(ycoarse)
```
I get:
```
WARNING: Method definition f(Any) in module Main at In[2]:4 overwritten at In[3]:4.

LoadError: MethodError: no method matching interpolate(::Array{Float64,1})
Closest candidates are:
  interpolate{IT<:Union{Interpolations.BSpline{D<:Interpolations.Degree},Interpolations.NoInterp,Tuple{Vararg{Union{Interpolations.BSpline,Interpolations.NoInterp},N}}},GT<:Union{Interpolations.GridType,Interpolations.NoInterp,Tuple{Vararg{Union{Interpolations.GridType,Interpolations.NoInterp},N}}}}(::AbstractArray{T,N}, !Matched::IT<:Union{Interpolations.BSpline{D<:Interpolations.Degree},Interpolations.NoInterp,Tuple{Vararg{Union{Interpolations.BSpline,Interpolations.NoInterp},N}}}, !Matched::GT<:Union{Interpolations.GridType,Interpolations.NoInterp,Tuple{Vararg{Union{Interpolations.GridType,Interpolations.NoInterp},N}}}) at /home/febbo/.julia/v0.5/Interpolations/src/b-splines/b-splines.jl:57
  interpolate{TWeights,TC,IT<:Union{Interpolations.BSpline{D<:Interpolations.Degree},Interpolations.NoInterp,Tuple{Vararg{Union{Interpolations.BSpline,Interpolations.NoInterp},N}}},GT<:Union{Interpolations.GridType,Interpolations.NoInterp,Tuple{Vararg{Union{Interpolations.GridType,Interpolations.NoInterp},N}}}}(!Matched::Type{TWeights}, !Matched::Type{TC}, !Matched::Any, !Matched::IT<:Union{Interpolations.BSpline{D<:Interpolations.Degree},Interpolations.NoInterp,Tuple{Vararg{Union{Interpolations.BSpline,Interpolations.NoInterp},N}}}, !Matched::GT<:Union{Interpolations.GridType,Interpolations.NoInterp,Tuple{Vararg{Union{Interpolations.GridType,Interpolations.NoInterp},N}}}) at /home/febbo/.julia/v0.5/Interpolations/src/b-splines/b-splines.jl:53
  interpolate{TWeights,TCoefs,Tel,N,IT<:Union{Interpolations.Gridded{D<:Interpolations.Degree},Interpolations.NoInterp,Tuple{Vararg{Union{Interpolations.Gridded,Interpolations.NoInterp},N}}}}(!Matched::Type{TWeights}, !Matched::Type{TCoefs}, !Matched::Tuple{Vararg{Union{AbstractArray{T,1},Tuple},N}}, !Matched::AbstractArray{Tel,N}, !Matched::IT<:Union{Interpolations.Gridded{D<:Interpolations.Degree},Interpolations.NoInterp,Tuple{Vararg{Union{Interpolations.Gridded,Interpolations.NoInterp},N}}}) at /home/febbo/.julia/v0.5/Interpolations/src/gridded/gridded.jl:52
  ...
while loading In[3], in expression starting on line 12
```
Any ideas?

**Comment by ChrisRackauckas** [comment]
I just tried it and got the same error.

**Comment by ChrisRackauckas** [comment]
The docs the example was taken from were just old. Using the README function:

```julia
itp = interpolate(A, BSpline(Linear()), OnGrid())
```

we got this working.

**Comment by timholy** [comment]
Where did you see these outdated docs?

**Comment by ChrisRackauckas** [comment]
https://github.com/tlycken/Interpolations.jl/blob/master/doc/Interpolations.jl.ipynb

Right at the start.

**Comment by timholy** [comment]
Fixed in #226
